lib/swaps/Swaps.ts

```diff
@@ -233,10 +290,89 @@ class Swaps extends EventEmitter {
       createTime: Date.now(),
     };
 
+    // add the deal. Going forward we can "record" errors related to this deal.
     this.addDeal(deal);
 
+    // verify LND setup. Make sure we are connected to BTC and LTC and that
+    // the peer is also connected to these networks.
+    const err = this.verifyLNDSetup(deal, peer);
+    if (err) {
+      this.setDealState(deal, SwapDealState.Error, err);
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
+      return;
+    }
+
+    const heldAmount = this.checkAndHoldAmount(deal);
+    assert(heldAmount === 0 || heldAmount === deal.takerAmount,
+        'Partial amount is not yet supported');
+    if (heldAmount === 0) {
+      this.setDealState(deal, SwapDealState.Error, 'Can not swap. No capacity is available');
```

Comment:
"Insufficient capacity" would be more accurate right?

Reply:
not really.

This case is really "no capacity". If there is some capacity we should try a reduced amount.

Reply:
OK I was thinking this referred to the lightning routing capacity but for available order quantity yes that is correct, nm.